src/typecheck.rs

```diff
@@ -81,6 +85,9 @@ impl lang::InlineOperator {
                         regex,
                         fields,
                         input_column.map(|e| e.into()),
+                        operator::ParseOptions {
+                            drop_nonmatching: options.contains(&lang::ParseOption::NoDrop)
```

Comment:
Is this condition backwards?  drop_nonmatching will be true if there's a "nodrop" option.